sdk/core/azure-core/src/main/java/com/azure/core/http/policy/DefaultRedirectStrategy.java

```diff
@@ -94,8 +95,10 @@ && isValidRedirectCount(tryCount)
             && isAllowedRedirectMethod(httpResponse.getRequest().getHttpMethod())) {
             String redirectUrl = tryGetRedirectHeader(httpResponse.getHeaders(), getLocationHeader());
             if (redirectUrl != null && !alreadyAttemptedRedirectUrl(redirectUrl, attemptedRedirectUrls)) {
-                LOGGER.verbose("[Redirecting] Try count: {}, Attempted Redirect URLs: {}", tryCount,
-                    attemptedRedirectUrls.toString());
+                LOGGER.atVerbose()
+                    .addKeyValue(LoggingKeys.TRY_COUNT_KEY, tryCount)
+                    .addKeyValue(LoggingKeys.REDIRECT_URL_KEY, redirectUrl)
```

Comment:
This no longer logs all the redirect URLs that were attempted, is this a logging behavior we want to change?

Reply:
good catch! I thought that Set.toString() would not work, but it does